src/custom/state/price/updater.ts

```diff
@@ -158,27 +158,20 @@ export default function FeesUpdater(): null {
       const refetchPrice = !unsupportedToken && priceIsOld(quoteInfo)
 
       if (unsupportedNeedsCheck || refetchAll || refetchPrice) {
-        let quoteData: Pick<QuoteInformationObject, 'chainId' | 'sellToken'> | undefined = undefined
-        // did the trade parameters change? major update required
-        if (quoteInfo && !quoteUsingSameParameters(quoteParams, quoteInfo)) {
-          quoteData = { chainId, sellToken }
-        }
-
-        // here we can set the loading quote state differently
-        // if params changed we signal a major update has happened by passing quoteData
-        // which in state will wipe the price
-        setLoadingQuote({
-          loading: true,
-          quoteData
-        })
+        const shouldFetchNewQuote = quoteInfo && !quoteUsingSameParameters(quoteParams, quoteInfo)
 
         refetchQuote({
           quoteParams,
           fetchFee: refetchAll,
-          previousFee: quoteInfo?.fee
-        })
-          .catch(error => console.error('Error re-fetching the quote', error))
-          .finally(() => setLoadingQuote({ loading: false }))
+          previousFee: quoteInfo?.fee,
+          handlers: {
+            handleSetLoadCb: () =>
```

Comment:
use handlers instead of using hook directly in useRefetch as the determination logic (soft vs hard quote load) is already done here

Reply:
nit: why to verbs? isn't `setLoadingCallback` good enough?